src/otx/backend/openvino/models/base.py

```diff
@@ -0,0 +1,346 @@
+        model = self.model.model if isinstance(self.model, Tiler) else self.model
+        resized_image = model.resize(image, (model.w, model.h))
+        resized_image = model.input_transform(resized_image)
+        return model._change_layout(resized_image)  # noqa: SLF001
+
+    def _read_ptq_config_from_ir(self, ov_model: Model) -> dict[str, Any]:
+        """Generates the PTQ (Post-Training Quantization) configuration from the meta data of the given OpenVINO model.
+
+        Args:
+            ov_model (Model): The OpenVINO model in which the PTQ configuration is embedded.
+
+        Returns:
+            dict: The PTQ configuration as a dictionary.
+        """
+        from nncf import IgnoredScope  # type: ignore[attr-defined]
+        from nncf.common.quantization.structs import QuantizationPreset  # type: ignore[attr-defined]
+        from nncf.parameters import ModelType
+        from nncf.quantization.advanced_parameters import AdvancedQuantizationParameters
+
+        if "optimization_config" not in ov_model.rt_info["model_info"]:
+            return {}
+
+        initial_ptq_config = json.loads(ov_model.rt_info["model_info"]["optimization_config"].value)
+        if not initial_ptq_config:
+            return {}
+        argparser = ArgumentParser()
+        if "advanced_parameters" in initial_ptq_config:
+            argparser.add_class_arguments(AdvancedQuantizationParameters, "advanced_parameters")
+        if "preset" in initial_ptq_config:
+            initial_ptq_config["preset"] = QuantizationPreset(initial_ptq_config["preset"])
+            argparser.add_argument("--preset", type=QuantizationPreset)
+        if "model_type" in initial_ptq_config:
+            initial_ptq_config["model_type"] = ModelType(initial_ptq_config["model_type"])
+            argparser.add_argument("--model_type", type=ModelType)
+        if "ignored_scope" in initial_ptq_config:
+            argparser.add_class_arguments(IgnoredScope, "ignored_scope", as_positional=True)
+
+        initial_ptq_config = argparser.parse_object(initial_ptq_config)
+
+        return argparser.instantiate_classes(initial_ptq_config).as_dict()
+
+    def prepare_metric_inputs(
+        self,
+        preds: OTXPredBatch,
+        inputs: OTXDataBatch,
+    ) -> MetricInput:
+        """Convert prediction and input entities to a format suitable for metric computation.
+
+        Args:
+            preds (OTXPredBatch): The predicted batch entity containing predicted labels.
+            inputs (OTXDataBatch): The input batch entity containing ground truth labels.
+
+        Returns:
+            MetricInput: A dictionary contains 'preds' and 'target' keys
+            corresponding to the predicted and target labels for metric evaluation.
+        """
+        raise NotImplementedError
+
+    def compute_metrics(self, meter: Metric) -> dict:
+        """Compute metrics using the provided torchvision Metric."""
+        return self._compute_metrics(meter)
```

Comment:
How about renaming `meter` to `metric`?
```suggestion
    def compute_metrics(self, metric: Metric) -> dict:
        """Compute metrics using the provided torchvision Metric."""
        return self._compute_metrics(metric)
```

Reply:
[Done](https://github.com/open-edge-platform/training_extensions/pull/4374/commits/0fd1d02c0cb2f4161ca017a2fd014f7bc717f6df)